spec/support/factory/formations/squad_request_factory.ex

```diff
@@ -11,7 +11,8 @@ defmodule Squadster.Support.Factory.Formations.SquadRequestFactory do
       end
 
       def approved(request) do
-        %{request | approver: insert(:squad_member), approved_at: Faker.DateTime.backward(10)}
+        approver = build(:squad_member, squad: nil, squad_id: request.squad_id)
```

Comment:
And also, what is the meaning of all these points, where you replace `insert` with `build`

Reply:
because with inserts when we `build(:squad) |> with_commander(commander)` it will build squad and **insert**  the commander, but it should insert the commander only when we pass all the construction to the `|> insert` function.